avocado/utils/configure_network.py

```diff
@@ -35,111 +35,176 @@ class NWException(Exception):
+                network_conf.write("IPV6_DEFROUTE=yes")
+        self.interface_up()
+
+    def unset_ip(self):
+        """
+        Unassign IP to Defined Interface
+        """
+        conf_file = ''
+        if distro.detect().name == 'rhel':
+            conf_file = "/etc/sysconfig/network-scripts/ifcfg-%s" % self._interface
+
+        if distro.detect().name == 'SuSE':
+            conf_file = "/etc/sysconfig/network/ifcfg-%s" % self._interface
+
+        self.interface_down()
+        self._move_interfaceconfile("%s.backup" % conf_file, conf_file)
+
+    def ping_check(self, peer_ip, count, option=None, flood=False):
+        """
+        Checks if the ping to peer works.
+        :param peer_ip :  Peer IP address
+        :param count   :  ping count
+        :param option  :  Default is None
+        :param flood   :  Default is False
+        :return        :  True / False if ping success
+        """
+        cmd = "ping -I %s %s -c %s" % (self._interface, peer_ip, count)
+        if flood:
+            cmd = "%s -f" % cmd
+        elif option:
+            cmd = "%s %s" % (cmd, option)
+        if process.system(cmd, shell=True, verbose=True,
+                          ignore_status=True) != 0:
+            return False
+        return True
+
+    def set_mtu_host(self, mtu):
+        """
+        Set MTU size in host interface
+        :param mtu :  mtu size that meed to be set
+        :return : return True / False in case of mtu able to set
+        """
+        cmd = "ip link set %s mtu %s" % (self._interface, mtu)
         try:
-            process.system(cmd, ignore_status=False, sudo=True)
-            return True
+            process.system(cmd, shell=True)
         except process.CmdError as ex:
-            raise NWException("ifup fails: %s" % ex)
+            raise NWException("MTU size can not be set: %s" % ex)
+        try:
+            cmd = "ip add show %s" % self._interface
+            mtuvalue = process.system_output(cmd, shell=True).decode("utf-8") \
+                .split()[4]
+            if mtuvalue == mtu:
+                return True
+        except Exception as ex:  # pylint: disable=W0703
+            raise NWException("setting MTU value in host failed: %s", ex)
+        return False
 
-    if distro.detect().name == 'SuSE':
-        conf_file = "/etc/sysconfig/network/ifcfg-%s" % interface
-        if os.path.exists(conf_file):
-            shutil.move(conf_file, conf_file+".backup")
-        else:
-            raise NWException("%s interface not available" % interface)
-        with open(conf_file, "w") as network_conf:
-            network_conf.write("IPADDR=%s \n" % ipaddr)
-            network_conf.write("NETMASK='%s' \n" % netmask)
-            network_conf.write("IPV6INIT=yes \n")
-            network_conf.write("IPV6_AUTOCONF=yes \n")
-            network_conf.write("IPV6_DEFROUTE=yes")
-
-        cmd = "ifup %s" % interface
+    def is_interface_link_up(self):
```

Comment:
we are already on a class named "NetworkInterface", so my suggestion is to rename this to `is_link_up`, so an usage example would be cleaner:

```python3
interface = NetworkInterface('lo')
interface.is_link_up()
```

Reply:
ah yes make sense